### PR TITLE
Add initial support for Lenovo A6000 (wt86518) and intruduce common dtsi for wt86518 and wt86528

### PR DESCRIPTION
**Pull request by jenneron**

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8-common.dtsi [hunk 1]
```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "msm8916-pm8916.dtsi"
+#include "msm8916-no-psci.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pinctrl/qcom,pmic-mpp.h>
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8-common.dtsi:6]: This seems unused so far

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include "msm8916-wingtech-wt865x8-common.dtsi"
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts:5]:
```suggestion
#include "msm8916-wingtech-wt865x8.dtsi"
```

I guess the x and .dtsi is sufficient to make it clear that this is some common thing, perhaps we can keep the filename a bit shorter.

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+/dts-v1/;
+
+#include "msm8916-wingtech-wt865x8-common.dtsi"
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "Lenovo A6010 (Wingtech WT86528)";
+	compatible = "wingtech,wt86528", "qcom,msm8916";
+
+	gpio-leds {
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts:12]: a6000 seems to also have some LEDs, although on different GPIOs. Although they don't seem to exist in the schematics... Have you tried if they work?

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+/ {
+	model = "Lenovo A6000 (Wingtech WT86518)";
+	compatible = "wingtech,wt86518", "qcom,msm8916";
+
+	usb_vbus: extcon-usb-dummy {
+		compatible = "linux,extcon-usb-dummy";
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts:12]: I assume you tested that GPIO110 doesn't work for USB ID on a6000?

**Review by Minecrell — COMMENTED** [review]
Perhaps add something like

```
A6000 and A6010 are quite similar, so some configuration is shared
in msm8916-wingtech-wt865x8.dtsi.
```

to the commit message.

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts (hunk 3 above)
Comment (jenneron) [reply arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts:12]: I don't see working leds with this configuration. https://dpaste.com/GZBP5BBW7

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts (hunk 4 above)
Comment (jenneron) [reply arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts:12]: Yes. I tested it with OTG adapter providing external power, which works with A6010.

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8-common.dtsi (hunk 1 above)
Comment (jenneron) [reply arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8-common.dtsi:6]: Okay. I guess i should delete `#include <dt-bindings/leds/common.h>` from wt86528 config too

Comment (Minecrell) [reply arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8-common.dtsi:6]: you have the gpio-leds there iirc

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts (hunk 4 above)
Comment (Minecrell) [reply arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts:12]: Did you just check if it's detected or also the status of the gpio in `/sys/kernel/debug/gpio`?

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8-common.dtsi (hunk 1 above)
Comment (jenneron) [reply arch/arm64/boot/dts/qcom/msm8916-wingtech-wt865x8-common.dtsi]: Nothing from this include is used. If you want i can return it again.

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts (hunk 4 above)
Comment (jenneron) [reply arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts:12]: Peripherals don't work with nothing in dmesg. I'll check debugfs

Comment (jenneron) [reply arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86518.dts:12]: The state is always same: `gpio110 : in high func0 8mA pull up`

**Review by Minecrell — APPROVED** [review]

**Pull request merged**